pkg/kubelet/util/util_windows.go

```diff
@@ -123,3 +130,41 @@ func GetBootTime() (time.Time, error) {
 	}
 	return currentTime.Add(-time.Duration(output) * time.Millisecond).Truncate(time.Second), nil
 }
+
+type reparseDataBufferHeader struct {
+	ReparseTag        uint32
+	ReparseDataLength uint16
+	Reserved          uint16
+}
+
+type reparseDataBuffer struct {
+	Header reparseDataBufferHeader
+	Detail [syscall.MAXIMUM_REPARSE_DATA_BUFFER_SIZE]byte
+}
+
+// IsUnixDomainSocket returns whether a given file is a AF_UNIX socket file
+func IsUnixDomainSocket(filePath string) (bool, error) {
+	// Due to the absence of golang support for os.ModeSocket in Windows (https://github.com/golang/go/issues/33357)
+	// we need to get the Reparse Points (https://docs.microsoft.com/en-us/windows/win32/fileio/reparse-points)
+	// for the file (using FSCTL_GET_REPARSE_POINT) and check for reparse tag: reparseTagSocket
+	fd, err := windows.CreateFile(windows.StringToUTF16Ptr(filePath), windows.GENERIC_READ, 0, nil, windows.OPEN_EXISTING, windows.FILE_FLAG_OPEN_REPARSE_POINT|windows.FILE_FLAG_BACKUP_SEMANTICS, 0)
```

Comment:
Would `CreateFile` leave any side effect?

Reply:
I see a closeHandle call, does it also require a delete or remove file call?

Reply:
`CreateFile`, when invoked with `windows.OPEN_EXISTING` (as is the case here) returns a handle to an existing file that the deferred `CloseHandle` will clean up. No other side effects involved. We do not need to delete/remove anything resulting from the `CreateFile` call as no new file gets created since CreateFile is invoked with `windows.OPEN_EXISTING`. I will add a comment and explain this to make it clear.